### PR TITLE
[DRAFT] AI controlled mobs

### PR DESCRIPTION
**Pull request by ondrej008**
## About The Pull Request

prepares to add mobs that can be controlled by the AI, sort of like /tg/ shells

## Why It's Good For The Game

- [x] Movement of the controlled thing
- [x] Smooth tracking of the controlled thing
- [ ] Changing to a camera or double clicking stops you from controlling or doesn't do anything.
- [x] Controlled thing acts like a camera
- [ ] AI actions (other than disconnect, probably) are gone.
- [ ] AI buttons for shells
- [ ] icons for ^
- [ ] ? AI verbs gone ?

## Changelog
:cl:
add: AI controlled mobs
/:cl:

### CODE REVIEW

File: code/_globalvars/lists/mobs.dm [hunk 1]
```diff
@@ -27,10 +27,11 @@ GLOBAL_LIST_EMPTY(mob_living_list)				//all instances of /mob/living and subtype
 GLOBAL_LIST_EMPTY(alive_living_list)		//all alive /mob/living, including clientless.
 GLOBAL_LIST_EMPTY(carbon_list)				//all instances of /mob/living/carbon and subtypes, notably does not contain brains or simple animals
 GLOBAL_LIST_EMPTY(offered_mob_list)				//all /mobs offered by admins
-GLOBAL_LIST_EMPTY(ai_list)
+GLOBAL_LIST_EMPTY(ai_list)					// i'm guessing this is the list of all ais
```
Comment (DominikPanic) [inline code/_globalvars/lists/mobs.dm]: Don't add useless comments like this

File: code/modules/mob/living/silicon/controlled/controlled.dm [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	for(var/mob/living/silicon/ai/ai in GLOB.ai_list)
+		to_chat(ai, "AI shell [name] created.")
+	GLOB.aiShells += src
+	
+
+/mob/living/silicon/controlled/Destroy()
+	for(var/mob/living/silicon/ai/ai in GLOB.ai_list)
+		to_chat(ai, "AI shell [name] destroyed.")
+	GLOB.aiShells -= src
+	..()
```
Comment (DominikPanic) [inline code/modules/mob/living/silicon/controlled/controlled.dm]:
```suggestion
	return ..()
```

File: code/modules/mob/living/silicon/controlled/controlled.dm [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+/mob/living/silicon/controlled/Initialize()
+	. = ..()
+	for(var/mob/living/silicon/ai/ai in GLOB.ai_list)
+		to_chat(ai, "AI shell [name] created.")
+	GLOB.aiShells += src
+	
+
+/mob/living/silicon/controlled/Destroy()
+	for(var/mob/living/silicon/ai/ai in GLOB.ai_list)
```
Comment (DominikPanic) [inline code/modules/mob/living/silicon/controlled/controlled.dm]: You can optimize this loop

File: code/modules/mob/living/silicon/controlled/controlled.dm [hunk 4]
```diff
@@ -0,0 +1,28 @@
+/mob/living/silicon/controlled
+	name = "meme man"
+	icon = 'icons/mob/human.dmi'
+	icon_state = "body_m_s"
+	var/move_delay = 1
+
+/mob/living/silicon/controlled/Initialize()
+	. = ..()
+	for(var/mob/living/silicon/ai/ai in GLOB.ai_list)
```
Comment (DominikPanic) [inline code/modules/mob/living/silicon/controlled/controlled.dm]: Same

File: code/modules/mob/living/silicon/ai/ai_verbs.dm [hunk 5]
```diff
@@ -315,3 +315,36 @@
...
+		remote_control = null
+
+	if(remote_control)
+		message_admins("Something went very bad here. [remote_control.name]")
+		return
+
+
+	var/list/possible = list()
+	
+	for(var/mob/living/silicon/controlled/borg in GLOB.aiShells)
```
Comment (DominikPanic) [inline code/modules/mob/living/silicon/ai/ai_verbs.dm]: Same

**Comment by spookydonut** [comment]
Stale closing

**Pull request closed without merging**